File: src/main/java/org/gangel/graphomance/arangodb/ArangoIdentifier.java
package org.gangel.graphomance.arangodb;

import lombok.Builder;
import lombok.Getter;
import org.gangel.graphomance.NodeIdentifier;
import org.gangel.graphomance.RelationIdentifier;

import java.util.Objects;

@Builder
public class ArangoIdentifier implements NodeIdentifier, RelationIdentifier {

    @Getter
    private String key;

    @Getter
    private String id;

    public static String getId(NodeIdentifier nodeId) {
        ArangoIdentifier idObj = Objects.requireNonNull((ArangoIdentifier) nodeId);
        return idObj.getId();
    }

    public static String getKey(NodeIdentifier nodeId) {
        ArangoIdentifier idObj = Objects.requireNonNull((ArangoIdentifier) nodeId);
        return idObj.getKey();
    }

    public static String getId(RelationIdentifier edgeId) {
        ArangoIdentifier idObj = Objects.requireNonNull((ArangoIdentifier) edgeId);
        return idObj.getId();
    }

    public static String getKey(RelationIdentifier edgeId) {
        ArangoIdentifier idObj = Objects.requireNonNull((ArangoIdentifier) edgeId);
        return idObj.getKey();
    }

}

File: src/main/kotlin/org/gangel/graphomance/arangodb/ArangoIdentifier.kt
package org.gangel.graphomance.arangodb

import org.gangel.graphomance.NodeIdentifier
import org.gangel.graphomance.RelationIdentifier
import java.util.Objects

class ArangoIdentifier(
	val key: String,
	val id: String
) : NodeIdentifier, RelationIdentifier {

	companion object {
		fun getId(nodeId: NodeIdentifier): String {
			val idObj: ArangoIdentifier = Objects.requireNonNull(nodeId as ArangoIdentifier)
			return idObj.id
		}

		fun getKey(nodeId: NodeIdentifier): String {
			val idObj: ArangoIdentifier = Objects.requireNonNull(nodeId as ArangoIdentifier)
			return idObj.key
		}

		fun getId(edgeId: RelationIdentifier): String {
			val idObj: ArangoIdentifier = Objects.requireNonNull(edgeId as ArangoIdentifier)
			return idObj.id
		}

		fun getKey(edgeId: RelationIdentifier): String {
			val idObj: ArangoIdentifier = Objects.requireNonNull(edgeId as ArangoIdentifier)
			return idObj.key
		}
	}
}